src/ring_theory/roots_of_unity.lean

```diff
@@ -327,6 +327,10 @@ begin
   { rintro rfl, exact one }
 end
 
+@[simp] lemma coe_submonoid_class_iff {M B : Type*} [comm_monoid M] [set_like B M]
```

Comment:
@Vierkantor Can you please check this is the intended way of using `submonoid_class`? Also, I am not able to use this new lemma to prove `coe_units_iff`, but maybe this is my fault. Thanks!

Reply:
This looks good to me!

The reason that `coe_units_iff` doesn't work is that apparently `units` is a structure, not a `subgroup` (presumably so that the inverse operator stays computable). In particular, the `units.ext` lemma is incompatible with `set_like.ext`.